ChainRouter.php

```diff
@@ -255,42 +255,32 @@ public function generate($name, $parameters = array(), $absolute = UrlGeneratorI
      *
      * If the request context is not set, this simply returns the request object built from $uri.
      *
-     * @param string $uri
+     * @param string $pathinfo
      *
      * @return Request
      */
-    private function rebuildRequest($uri)
+    private function rebuildRequest($pathinfo)
     {
         if (!$this->context) {
-            return Request::create($uri);
+            return Request::create('http://localhost'.$pathinfo);
         }
 
+        $uri = $pathinfo;
+
         $server = array();
-        if ($this->context->getHost()) {
-            $server['SERVER_NAME'] = $this->context->getHost();
-            $server['HTTP_HOST'] = $this->context->getHost();
-        }
         if ($this->context->getBaseUrl()) {
-            $uri = $this->context->getBaseUrl().$uri;
+            $uri = $this->context->getBaseUrl().$pathinfo;
             $server['SCRIPT_FILENAME'] = $this->context->getBaseUrl();
             $server['PHP_SELF'] = $this->context->getBaseUrl();
         }
-        if ('https' === $this->context->getScheme()) {
-            $server['HTTPS'] = 'on';
-            $server['SERVER_PORT'] = $this->context->getHttpsPort();
-            if (443 !== $this->context->getHttpsPort()) {
-                // this is parsed from the host by symfony request
-                // https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L971
-                $server['HTTP_HOST'] .= ':'.$this->context->getHttpsPort();
-            }
-        } else {
-            $server['SERVER_PORT'] = $this->context->getHttpPort();
-            if (80 !== $this->context->getHttpPort()) {
-                // this is parsed from the host by symfony request
-                // https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L971
-                $server['HTTP_HOST'] .= ':'.$this->context->getHttpPort();
-            }
+        $host = $this->context->getHost() ?: 'localhost';
+        if ('https' === $this->context->getScheme() && 443 !== $this->context->getHttpsPort()) {
+            $host .= ':'.$this->context->getHttpsPort();
+        }
+        if ('http' === $this->context->getScheme() && 80 !== $this->context->getHttpPort()) {
+            $host .= ':'.$this->context->getHttpPort();
         }
+        $uri = $this->context->getScheme().'://'.$host.$uri.$this->context->getQueryString();
```

Comment:
this does not work. it's missing the `?` before the query string. this is why i linked the exmaple implementation https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L1067